artifactory/commands/terraform/terraformpublish.go

```diff
@@ -0,0 +1,365 @@
+	"github.com/jfrog/jfrog-client-go/artifactory/services"
+	clientservicesutils "github.com/jfrog/jfrog-client-go/artifactory/services/utils"
+	specutils "github.com/jfrog/jfrog-client-go/artifactory/services/utils"
+	clientutils "github.com/jfrog/jfrog-client-go/utils"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"github.com/jfrog/jfrog-client-go/utils/io/content"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"io"
+	"io/fs"
+	"os"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+type TerraformPublishCommandArgs struct {
+	namespace  string
+	moduleName string
+	provider   string
+	tag        string
+	exclusions []string
+}
+
+type TerraformPublishCommand struct {
+	*TerraformPublishCommandArgs
+	args           []string
+	repo           string
+	configFilePath string
+	serverDetails  *config.ServerDetails
+	result         *commandsutils.Result
+}
+
+func NewTerraformPublishCommand() *TerraformPublishCommand {
+	return &TerraformPublishCommand{TerraformPublishCommandArgs: NewTerraformPublishCommandArgs(), result: new(commandsutils.Result)}
+}
+
+func NewTerraformPublishCommandArgs() *TerraformPublishCommandArgs {
+	return &TerraformPublishCommandArgs{}
+}
+
+func (tpc *TerraformPublishCommand) GetArgs() []string {
+	return tpc.args
+}
+
+func (tpc *TerraformPublishCommand) SetArgs(terraformArg []string) *TerraformPublishCommand {
+	tpc.args = terraformArg
+	return tpc
+}
+
+func (tpc *TerraformPublishCommand) SetServerDetails(serverDetails *config.ServerDetails) *TerraformPublishCommand {
+	tpc.serverDetails = serverDetails
+	return tpc
+}
+
+func (tpc *TerraformPublishCommand) SetRepo(repo string) *TerraformPublishCommand {
+	tpc.repo = repo
+	return tpc
+}
+
+func (tpc *TerraformPublishCommand) ServerDetails() (*config.ServerDetails, error) {
+	return tpc.serverDetails, nil
+}
+
+func (tpc *TerraformPublishCommand) CommandName() string {
+	return "rt_terraform_publish"
+}
+
+func (tpc *TerraformPublishCommand) SetConfigFilePath(configFilePath string) *TerraformPublishCommand {
+	tpc.configFilePath = configFilePath
+	return tpc
+}
+
+func (tpc *TerraformPublishCommand) Result() *commandutils.Result {
+	return tpc.result
+}
+
+func (tpc *TerraformPublishCommand) SetModuleName(moduleName string) *TerraformPublishCommand {
+	tpc.moduleName = moduleName
+	return tpc
+}
+
+func (tpc *TerraformPublishCommand) Run() error {
+	log.Info("Running Terraform publish")
+	err := tpc.preparePrerequisites()
+	if err != nil {
+		return err
+	}
+	err = tpc.publish()
+	if err != nil {
+		return err
+	}
+	log.Info("Terraform publish finished successfully.")
+	return nil
+}
+
+func (tpc *TerraformPublishCommand) preparePrerequisites() error {
+	err := tpc.extractTerraformPublishOptionsFromArgs(tpc.args)
+	if err != nil {
+		return err
+	}
+	if tpc.namespace == "" || tpc.provider == "" || tpc.tag == "" {
+		return errorutils.CheckErrorf("Wrong number of arguments. for a terraform publish command please provide --namespace, --provider and --tag.")
```

Comment:
Wrong number of arguments. for a terraform publish command please provide --namespace, --provider and --tag.
-->
The --namespace, --provider and --tag options are mandatory.